### ISSUE TITLE
Indentation level set by already-closed brackets

### ISSUE THREAD

**Issue by kbrose** [DSpeckhals/python-indent#7]
It seems like the indentation level will set itself to the most recent open bracket, even if it has already been closed. Consider:

``` python
x = [[0, 1, 2], [3, 4, 5],<cursor here>]
```

turns into

``` python
x = [[0, 1, 2], [3, 4, 5],
                 <cursor here>]]
```

instead of

``` python
x = [[0, 1, 2], [3, 4, 5],
     <cursor here>]
```

as expected.

**version:**
Using python-indent version 0.3.3 (specifically, without commit 6d7adb5eaeb1e1824e212a887de8a903fc87dd82 that fixed #5)

**Comment by DSpeckhals** [comment]
Ah yes. I've known this problem exists, but it actually will require some interesting thought. Right now, this package does not match brackets; it just looks for the last open delimiter, which is a fundamental road block.

I'd love to make this work. I have a few ideas, including using some of Atom's builtin bracket-matching logic, and/or somehow keeping track of which delimiters are open and closed. I'll take a look this week, and let you know what I find. Thanks for the detailed report!

**Comment by kbrose** [comment]
Yeah - I realized that this would be a major problem after I posted it. Probably not possible with the current regex approach.

I was also planning on looking into it. My first thought was doing our own bracket counting (being sure to ignore brackets in comments and strings), but if Atom has an API for that, that'd be great too.

**Comment by DSpeckhals** [comment]
This was fixed with #9.
